chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -566,6 +602,37 @@ impl PartialWitnessActor {
         validators.sort();
         Ok(validators)
     }
+
+    /// Returns for each code-hash the contract code retrieved from the storage.
+    fn retrieve_contract_code<'a, I>(
+        &self,
+        key: &ChunkProductionKey,
+        code_hashes: I,
+    ) -> Result<Vec<CodeBytes>, Error>
+    where
+        I: Iterator<Item = &'a CodeHash>,
+    {
+        let storage = TrieDBStorage::new(
+            TrieStoreAdapter::new(self.runtime.store().clone()),
+            self.epoch_manager.shard_id_to_uid(key.shard_id, &key.epoch_id)?,
+        );
+        let mut contracts = Vec::new();
+        for contract_hash in code_hashes {
+            match storage.retrieve_raw_bytes(&contract_hash.0) {
+                Ok(bytes) => contracts.push(CodeBytes(bytes)),
+                Err(StorageError::MissingTrieValue(_, _)) => {
+                    tracing::warn!(
+                        target: "client",
+                        ?contract_hash,
+                        chunk_production_key = ?key,
+                        "Requested contract hash is not present in the storage"
+                    );
```

Comment:
note: you are changing the current behaviour: previously we didn't send the response on missing key.
that is OK, I was going to rework that part anyway, just FYI about the accidental change

Reply:
I had extracted this code because in a prior version I wanted to reuse this code. Also to address the other comment about the signature, I will inline this back in the only caller.

Reply:
done with reverting this. I will let you work on the change in that code.